scripts/src/java/org/oppia/android/scripts/testfile/TestFileCheck.kt

```diff
@@ -0,0 +1,85 @@
+package org.oppia.android.scripts.testfile
+
+import org.oppia.android.scripts.common.RepositoryFile
+import org.oppia.android.scripts.common.ScriptExemptions
+import org.oppia.android.scripts.common.TEST_FILE_CHECK_FAILED_OUTPUT_INDICATOR
+import org.oppia.android.scripts.common.TEST_FILE_CHECK_PASSED_OUTPUT_INDICATOR
+import java.io.File
+
+/**
+ * Script for ensuring that all production files have test files present.
+ *
+ * Usage:
+ *   bazel run //scripts:test_file_check -- <path_to_directory_root>
+ *
+ * Arguments:
+ * - path_to_directory_root: directory path to the root of the Oppia Android repository.
+ *
+ * Example:
+ *   bazel run //scripts:test_file_check -- $(pwd)
+ */
+fun main(vararg args: String) {
+  // Path of the repo to be analyzed.
+  val repoPath = args[0] + "/"
+
+  // A list of all kotlin files in the repo to be analyzed.
+  val searchFiles = RepositoryFile.collectSearchFiles(
+    repoPath = repoPath,
+    expectedExtension = ".kt",
+    exemptionsList = ScriptExemptions.TEST_FILE_CHECK_EXEMPTIONS_LIST
+  )
+
+  // A list of all the prod files present in the repo.
+  val prodFilesList = mutableListOf<File>()
+
+  // A list of all the test files present in the repo.
+  val testFilesList = mutableListOf<File>()
+
+  searchFiles.forEach { file ->
+    if (file.name.endsWith("Test.kt")) {
+      testFilesList.add(file)
```

Comment:
Ditto here (per a comment from an earlier PR): prefer creating the two lists here in a more Kotlin-idiomatic way.

Reply:
Done.